server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -1012,17 +1014,29 @@ private SortedMap<TServerInstance,TabletServerStatus> gatherTableInformation(
             badServers.remove(server);
           }
         }
-      });
-    }
-    tp.shutdown();
-    try {
-      tp.awaitTermination(Math.max(10000, rpcTimeout / 3), TimeUnit.MILLISECONDS);
-    } catch (InterruptedException e) {
-      log.debug("Interrupted while fetching status");
+      }));
+    }
+    long timeToWaitForCompletion = Math.max(10000, rpcTimeout / 3);
+    long currTime = System.currentTimeMillis();
```

Comment:
Do not rely on `System.currentTimeMillis()` for tracking duration of executions. That is wall clock time, and can be changed by NTP, arbitrarily by a user, a time zone change, daylight savings, leap time, etc. For timing duration of events inside Java, always use `System.nanoTime()`, not because you actually get nanosecond granularity (you probably don't), but because `nanoTime(t2) - nanoTime(t1)` has stronger guarantees.

Reply:
Resolved in e888bdc